src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1257,6 +1262,21 @@ class Config:
         datastore = 'vm.device'
         datastore_extend = 'vm.device.extend_device'
 
+    @private
+    async def nic_check(self, vm_device):
+        if not await self.middleware.call('system.is_freenas') and await self.middleware.call('failover.licensed'):
+            failover_enabled = not (await self.middleware.call('failover.config'))['disabled']
+            nics = await self.middleware.call('interface.vm_checks', [vm_device])
+            if nics:
+                if failover_enabled:
+                    raise CallError('Failover must be disabled before attempting to create/update VM NIC device')
+                else:
+                    for nic in nics:
+                        await self.middleware.call('interface.disable_capabilities', nic, nics[nic])
+                        await self.middleware.call(
+                            'failover.call_remote', 'interface.disable_capabilities', [nic, nics[nic]]
+                        )
```

Comment:
Standby node may be down, we need to handle that exception here and other places.

Reply:
Maybe this method `interface.disable_capabilities` should propagate itself to standby node?

Reply:
@themylogin there's still possibility for remote node to get out of sync with master node ( a case is remote node is down when this is called for master ). Please advise if we should still consider that ?